base/abstractarray.jl

```diff
@@ -3075,6 +3093,11 @@ julia> mapslices(f, A, dims=(1,2))
 [:, :, 3] =
  21  21  21  21
 
+julia> f(x::AbstractMatrix) = fill(x[1,1], 1,4);
+
+julia> B == stack(f, eachslice(A, dims=3))
+true
```

Comment:
Maybe this function should have a distinct name:
```suggestion
julia> f2(x::AbstractMatrix) = fill(x[1,1], 1,4);

julia> B == stack(f2, eachslice(A, dims=3))
true
```